controllers/todo.js

```diff
@@ -1,40 +1,173 @@
-const { ToDo, Token } = require("../models");
-
-// All the given method require token.
-// So be sure to check for it before doing any stuff
-// HINT: Create a middleware for above :)
+const { ToDo, User } = require("../models");
 
 const getAllToDo = async (req, res) => {
-  // Get the token in header.
-  // Use the token to get all the ToDo's of a user
+  const userId = req.user.id;
+
+  const createdTodos = await ToDo.find({ createdBy: userId });
+  const user = await User.findById(userId).populate('collaboratingTodos');
+  if (createdTodos) {
+    res.status(200).json({ createdTodos: createdTodos, collaboratingTodos: user.collaboratingTodos });
+  } else {
+    res.status(404).send("Todos not found");
+  }
 };
 
 const createToDo = async (req, res) => {
-  // Check for the token and create a todo
-  // or throw error correspondingly
+  const { title } = req.body;
+
+  if (!title) {
+    res.status(400).send("Title field is empty!");
+  } else {
+    const newTodo = new ToDo({
+      title: title,
+      createdBy: req.user.id
+    });
+    await newTodo.save();
+
+    res.status(200).json({ title: title, createdBy: req.user.id });
+  }
 };
 
+const creator = (userId, todo) => { return userId == todo.createdBy; }
+
+const collaborator = (userId, todo) => {
+  const foundCollaborator = todo.collaborators.filter(collaboratorId => collaboratorId == userId);
+  if (foundCollaborator.length > 0) return true;
+  return false;
+}
```

Comment:
Can be made better. Instead of filtering, just use the following
```js
const collaborator = (userId, todo) =>  todo.collaborators.includes(userId)
```

Reply:
Sweet trick! thanks for the info, will keep that in mind.